synapse/push/httppusher.py

```diff
@@ -358,80 +363,70 @@ async def _build_notification_dict(
+                "counts": {"unread": badge},
+                "prio": priority,
             }
-            return d
+            return content, {}
 
         ctx = await push_tools.get_context_for_event(
             self._storage_controllers, event, self.user_id
         )
 
-        d = {
-            "notification": {
-                "id": event.event_id,  # deprecated: remove soon
-                "event_id": event.event_id,
-                "room_id": event.room_id,
-                "type": event.type,
-                "sender": event.user_id,
-                "prio": priority,
-                "counts": {
-                    "unread": badge,
-                    # 'missed_calls': 2
-                },
-                "devices": [
-                    {
-                        "app_id": self.app_id,
-                        "pushkey": self.pushkey,
-                        "pushkey_ts": int(self.pushkey_ts / 1000),
-                        "data": self.data_minus_url,
-                        "tweaks": tweaks,
-                    }
-                ],
-            }
+        content = {
+            "id": event.event_id,  # deprecated: remove soon
+            "event_id": event.event_id,
+            "room_id": event.room_id,
+            "type": event.type,
+            "sender": event.user_id,
+            "prio": priority,
+            "counts": {
+                "unread": badge,
+                # 'missed_calls': 2
+            },
         }
         if event.type == "m.room.member" and event.is_state():
-            d["notification"]["membership"] = event.content["membership"]
-            d["notification"]["user_is_target"] = event.state_key == self.user_id
+            content["membership"] = event.content["membership"]
+            content["user_is_target"] = event.state_key == self.user_id
         if self.hs.config.push.push_include_content and event.content:
-            d["notification"]["content"] = event.content
+            content["content"] = event.content
 
         # We no longer send aliases separately, instead, we send the human
         # readable name of the room, which may be an alias.
         if "sender_display_name" in ctx and len(ctx["sender_display_name"]) > 0:
-            d["notification"]["sender_display_name"] = ctx["sender_display_name"]
+            content["sender_display_name"] = ctx["sender_display_name"]
         if "name" in ctx and len(ctx["name"]) > 0:
-            d["notification"]["room_name"] = ctx["name"]
+            content["room_name"] = ctx["name"]
+
+        return (content, tweaks)
+
+    def _build_notification_dict(
+        self, content: JsonDict, tweaks: Optional[JsonMapping]
+    ) -> JsonDict:
+        device = {
+            "app_id": self.app_id,
+            "pushkey": self.pushkey,
+            "pushkey_ts": int(self.pushkey_ts / 1000),
+            "data": self.data_minus_url,
+        }
+        if tweaks:
+            device["tweaks"] = tweaks
```

Comment:
This was previously not set for `event_id_only`, I think? This is a change in behavior.

Reply:
It will be clearer after adding some docstrings, but `_build_event_notification` takes `tweaks` and return `tweaks`, which will be empty for the `event_id_only` use case.